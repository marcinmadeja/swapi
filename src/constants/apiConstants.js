export const SWAPI_URL = 'https://swapi.co/api';
export const SWAPI_FILMS = '/films/?format=json';

export const RANDOM_USERS = 'https://randomuser.me/api';
