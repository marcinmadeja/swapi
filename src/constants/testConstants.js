/* eslint-disable */

export const users = [
  {
    gender: 'male',
    name: {
      title: 'mr',
      first: 'terrence',
      last: 'watts',
    },
    location: {
      street: '5796 rolling green rd',
      city: 'bowral',
      state: 'victoria',
      postcode: 7762,
      coordinates: {
        latitude: '-79.9438',
        longitude: '-160.0279',
      },
      timezone: {
        offset: '-7:00',
        description: 'Mountain Time (US & Canada)',
      },
    },
    email: 'terrence.watts@example.com',
    login: {
      uuid: 'ef30b2c2-7575-42d1-b9e5-38030d64b184',
      username: 'tinygoose275',
      password: 'makaveli',
      salt: 'EwEwiYhD',
      md5: '67600680d1f62c990021d597e96b264d',
      sha1: '174b563d3f42d98f88e7bcaad064e647ee726faa',
      sha256: 'fc93dd24693ad0650413b01980ccc494dd3921991a8ab41e45cadf8610aa046a',
    },
    dob: {
      date: '1972-09-08T12:37:28Z',
      age: 45,
    },
    registered: {
      date: '2003-03-25T20:55:50Z',
      age: 15,
    },
    phone: '00-8437-1526',
    cell: '0476-938-903',
    id: {
      name: 'TFN',
      value: '686111438',
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/men/24.jpg',
      medium: 'https://randomuser.me/api/portraits/med/men/24.jpg',
      thumbnail: 'https://randomuser.me/api/portraits/thumb/men/24.jpg',
    },
    nat: 'AU',
  },
  {
    gender: 'male',
    name: {
      title: 'mr',
      first: 'paulo',
      last: 'seyfarth',
    },
    location: {
      street: 'talstra√üe 94',
      city: 'bogen',
      state: 'bayern',
      postcode: 74635,
      coordinates: {
        latitude: '18.2360',
        longitude: '72.7154',
      },
      timezone: {
        offset: '-3:00',
        description: 'Brazil, Buenos Aires, Georgetown',
      },
    },
    email: 'paulo.seyfarth@example.com',
    login: {
      uuid: 'af5ee15a-da22-4264-8c94-7cd57e21bf97',
      username: 'crazybutterfly740',
      password: 'colt45',
      salt: '7LDAWnJY',
      md5: 'e5bbaee291478ce51472ec82d31f97c6',
      sha1: 'ea8722d0fc4ba69dc43e36661468f5bda5199799',
      sha256: 'e31a953387293ab446bf702c20013a91f1c91793fe59fe425a6e6502d502ad50',
    },
    dob: {
      date: '1969-06-18T22:13:53Z',
      age: 49,
    },
    registered: {
      date: '2004-09-15T22:48:16Z',
      age: 13,
    },
    phone: '0477-2629824',
    cell: '0175-1061186',
    id: {
      name: '',
      value: null,
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/men/98.jpg',
      medium: 'https://randomuser.me/api/portraits/med/men/98.jpg',
      thumbnail: 'https://randomuser.me/api/portraits/thumb/men/98.jpg',
    },
    nat: 'DE',
  },
  {
    gender: 'male',
    name: {
      title: 'mr',
      first: 'mathias',
      last: 'mortensen',
    },
    location: {
      street: '9165 fredensvej',
      city: 'pandrup',
      state: 'nordjylland',
      postcode: 39076,
      coordinates: {
        latitude: '-12.9436',
        longitude: '101.5652',
      },
      timezone: {
        offset: '+7:00',
        description: 'Bangkok, Hanoi, Jakarta',
      },
    },
    email: 'mathias.mortensen@example.com',
    login: {
      uuid: '39d341b7-2f7b-4b4a-bfd4-6921922da746',
      username: 'beautifulladybug715',
      password: 'carpente',
      salt: 'KpPfkc93',
      md5: '5f976528c474e2c1479464183d7800d9',
      sha1: '6d25ac15f23bf114e05cb940e7c61925b3274746',
      sha256: 'cf715b8dc50773efd203d755f2b0c268fd588e5b98204545ad1a71d30b66e4b2',
    },
    dob: {
      date: '1990-04-03T15:42:14Z',
      age: 28,
    },
    registered: {
      date: '2014-05-19T01:46:06Z',
      age: 4,
    },
    phone: '04223613',
    cell: '62185716',
    id: {
      name: 'CPR',
      value: '722632-7968',
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/men/13.jpg',
      medium: 'https://randomuser.me/api/portraits/med/men/13.jpg',
      thumbnail: 'https://randomuser.me/api/portraits/thumb/men/13.jpg',
    },
    nat: 'DK',
  },
];

export const swapiPeople = [
  {
    name: 'Leia Organa',
    height: '150',
    mass: '49',
    hair_color: 'brown',
    skin_color: 'light',
    eye_color: 'brown',
    birth_year: '19BBY',
    gender: 'female',
    homeworld: 'https://swapi.co/api/planets/2/',
    films: [
      'https://swapi.co/api/films/2/',
      'https://swapi.co/api/films/6/',
      'https://swapi.co/api/films/3/',
      'https://swapi.co/api/films/1/',
      'https://swapi.co/api/films/7/',
    ],
    species: [
      'https://swapi.co/api/species/1/',
    ],
    vehicles: [
      'https://swapi.co/api/vehicles/30/',
    ],
    starships: [],
    created: '2014-12-10T15:20:09.791000Z',
    edited: '2014-12-20T21:17:50.315000Z',
    url: 'https://swapi.co/api/people/5/',
  },
  {
    name: 'Obi-Wan Kenobi',
    height: '182',
    mass: '77',
    hair_color: 'auburn, white',
    skin_color: 'fair',
    eye_color: 'blue-gray',
    birth_year: '57BBY',
    gender: 'male',
    homeworld: 'https://swapi.co/api/planets/20/',
    films: [
      'https://swapi.co/api/films/2/',
      'https://swapi.co/api/films/5/',
      'https://swapi.co/api/films/4/',
      'https://swapi.co/api/films/6/',
      'https://swapi.co/api/films/3/',
      'https://swapi.co/api/films/1/',
    ],
    species: [
      'https://swapi.co/api/species/1/',
    ],
    vehicles: [
      'https://swapi.co/api/vehicles/38/',
    ],
    starships: [
      'https://swapi.co/api/starships/48/',
      'https://swapi.co/api/starships/59/',
      'https://swapi.co/api/starships/64/',
      'https://swapi.co/api/starships/65/',
      'https://swapi.co/api/starships/74/',
    ],
    created: '2014-12-10T16:16:29.192000Z',
    edited: '2014-12-20T21:17:50.325000Z',
    url: 'https://swapi.co/api/people/10/',
  },
];

export const swapiFilms = [
  {
    "title": "A New Hope",
    "episode_id": 4,
    "opening_crawl": "It is a period of civil war.\r\nRebel spaceships, striking\r\nfrom a hidden base, have won\r\ntheir first victory against\r\nthe evil Galactic Empire.\r\n\r\nDuring the battle, Rebel\r\nspies managed to steal secret\r\nplans to the Empire's\r\nultimate weapon, the DEATH\r\nSTAR, an armored space\r\nstation with enough power\r\nto destroy an entire planet.\r\n\r\nPursued by the Empire's\r\nsinister agents, Princess\r\nLeia races home aboard her\r\nstarship, custodian of the\r\nstolen plans that can save her\r\npeople and restore\r\nfreedom to the galaxy....",
    "director": "George Lucas",
    "producer": "Gary Kurtz, Rick McCallum",
    "release_date": "1977-05-25",
    "characters": [
      "https://swapi.co/api/people/1/",
      "https://swapi.co/api/people/2/",
      "https://swapi.co/api/people/3/",
      "https://swapi.co/api/people/4/",
      "https://swapi.co/api/people/5/",
      "https://swapi.co/api/people/6/",
      "https://swapi.co/api/people/7/",
      "https://swapi.co/api/people/8/",
      "https://swapi.co/api/people/9/",
      "https://swapi.co/api/people/10/",
      "https://swapi.co/api/people/12/",
      "https://swapi.co/api/people/13/",
      "https://swapi.co/api/people/14/",
      "https://swapi.co/api/people/15/",
      "https://swapi.co/api/people/16/",
      "https://swapi.co/api/people/18/",
      "https://swapi.co/api/people/19/",
      "https://swapi.co/api/people/81/"
    ],
    "planets": [
      "https://swapi.co/api/planets/2/",
      "https://swapi.co/api/planets/3/",
      "https://swapi.co/api/planets/1/"
    ],
    "starships": [
      "https://swapi.co/api/starships/2/",
      "https://swapi.co/api/starships/3/",
      "https://swapi.co/api/starships/5/",
      "https://swapi.co/api/starships/9/",
      "https://swapi.co/api/starships/10/",
      "https://swapi.co/api/starships/11/",
      "https://swapi.co/api/starships/12/",
      "https://swapi.co/api/starships/13/"
    ],
    "vehicles": [
      "https://swapi.co/api/vehicles/4/",
      "https://swapi.co/api/vehicles/6/",
      "https://swapi.co/api/vehicles/7/",
      "https://swapi.co/api/vehicles/8/"
    ],
    "species": [
      "https://swapi.co/api/species/5/",
      "https://swapi.co/api/species/3/",
      "https://swapi.co/api/species/2/",
      "https://swapi.co/api/species/1/",
      "https://swapi.co/api/species/4/"
    ],
    "created": "2014-12-10T14:23:31.880000Z",
    "edited": "2015-04-11T09:46:52.774897Z",
    "url": "https://swapi.co/api/films/1/"
  }
];

export const swapiStarships = [
  {
    "name": "Death Star",
    "model": "DS-1 Orbital Battle Station",
    "manufacturer": "Imperial Department of Military Research, Sienar Fleet Systems",
    "cost_in_credits": "1000000000000",
    "length": "120000",
    "max_atmosphering_speed": "n/a",
    "crew": "342953",
    "passengers": "843342",
    "cargo_capacity": "1000000000000",
    "consumables": "3 years",
    "hyperdrive_rating": "4.0",
    "MGLT": "10",
    "starship_class": "Deep Space Mobile Battlestation",
    "pilots": [],
    "films": [
      "https://swapi.co/api/films/1/"
    ],
    "created": "2014-12-10T16:36:50.509000Z",
    "edited": "2014-12-22T17:35:44.452589Z",
    "url": "https://swapi.co/api/starships/9/"
  }
];

export const swapiVehicles = [
  {
    "name": "Snowspeeder",
    "model": "t-47 airspeeder",
    "manufacturer": "Incom corporation",
    "cost_in_credits": "unknown",
    "length": "4.5",
    "max_atmosphering_speed": "650",
    "crew": "2",
    "passengers": "0",
    "cargo_capacity": "10",
    "consumables": "none",
    "vehicle_class": "airspeeder",
    "pilots": [
      "https://swapi.co/api/people/1/",
      "https://swapi.co/api/people/18/"
    ],
    "films": [
      "https://swapi.co/api/films/2/"
    ],
    "created": "2014-12-15T12:22:12Z",
    "edited": "2014-12-22T18:21:15.623033Z",
    "url": "https://swapi.co/api/vehicles/14/"
  }
];

export const swapiPlanets = [
  {
    "name": "Human",
    "classification": "mammal",
    "designation": "sentient",
    "average_height": "180",
    "skin_colors": "caucasian, black, asian, hispanic",
    "hair_colors": "blonde, brown, black, red",
    "eye_colors": "brown, blue, green, hazel, grey, amber",
    "average_lifespan": "120",
    "homeworld": "https://swapi.co/api/planets/9/",
    "language": "Galactic Basic",
    "people": [
      "https://swapi.co/api/people/1/",
      "https://swapi.co/api/people/4/",
      "https://swapi.co/api/people/5/",
      "https://swapi.co/api/people/6/",
      "https://swapi.co/api/people/7/",
      "https://swapi.co/api/people/9/",
      "https://swapi.co/api/people/10/",
      "https://swapi.co/api/people/11/",
      "https://swapi.co/api/people/12/",
      "https://swapi.co/api/people/14/",
      "https://swapi.co/api/people/18/",
      "https://swapi.co/api/people/19/",
      "https://swapi.co/api/people/21/",
      "https://swapi.co/api/people/22/",
      "https://swapi.co/api/people/25/",
      "https://swapi.co/api/people/26/",
      "https://swapi.co/api/people/28/",
      "https://swapi.co/api/people/29/",
      "https://swapi.co/api/people/32/",
      "https://swapi.co/api/people/34/",
      "https://swapi.co/api/people/43/",
      "https://swapi.co/api/people/51/",
      "https://swapi.co/api/people/60/",
      "https://swapi.co/api/people/61/",
      "https://swapi.co/api/people/62/",
      "https://swapi.co/api/people/66/",
      "https://swapi.co/api/people/67/",
      "https://swapi.co/api/people/68/",
      "https://swapi.co/api/people/69/",
      "https://swapi.co/api/people/74/",
      "https://swapi.co/api/people/81/",
      "https://swapi.co/api/people/84/",
      "https://swapi.co/api/people/85/",
      "https://swapi.co/api/people/86/",
      "https://swapi.co/api/people/35/"
    ],
    "films": [
      "https://swapi.co/api/films/2/",
      "https://swapi.co/api/films/7/",
      "https://swapi.co/api/films/5/",
      "https://swapi.co/api/films/4/",
      "https://swapi.co/api/films/6/",
      "https://swapi.co/api/films/3/",
      "https://swapi.co/api/films/1/"
    ],
    "created": "2014-12-10T13:52:11.567000Z",
    "edited": "2015-04-17T06:59:55.850671Z",
    "url": "https://swapi.co/api/species/1/"
  }
];

export const swapiSpecies = [
  {
    "name": "Human",
    "classification": "mammal",
    "designation": "sentient",
    "average_height": "180",
    "skin_colors": "caucasian, black, asian, hispanic",
    "hair_colors": "blonde, brown, black, red",
    "eye_colors": "brown, blue, green, hazel, grey, amber",
    "average_lifespan": "120",
    "homeworld": "https://swapi.co/api/planets/9/",
    "language": "Galactic Basic",
    "people": [
      "https://swapi.co/api/people/1/",
      "https://swapi.co/api/people/4/",
      "https://swapi.co/api/people/5/",
      "https://swapi.co/api/people/6/",
      "https://swapi.co/api/people/7/",
      "https://swapi.co/api/people/9/",
      "https://swapi.co/api/people/10/",
      "https://swapi.co/api/people/11/",
      "https://swapi.co/api/people/12/",
      "https://swapi.co/api/people/14/",
      "https://swapi.co/api/people/18/",
      "https://swapi.co/api/people/19/",
      "https://swapi.co/api/people/21/",
      "https://swapi.co/api/people/22/",
      "https://swapi.co/api/people/25/",
      "https://swapi.co/api/people/26/",
      "https://swapi.co/api/people/28/",
      "https://swapi.co/api/people/29/",
      "https://swapi.co/api/people/32/",
      "https://swapi.co/api/people/34/",
      "https://swapi.co/api/people/43/",
      "https://swapi.co/api/people/51/",
      "https://swapi.co/api/people/60/",
      "https://swapi.co/api/people/61/",
      "https://swapi.co/api/people/62/",
      "https://swapi.co/api/people/66/",
      "https://swapi.co/api/people/67/",
      "https://swapi.co/api/people/68/",
      "https://swapi.co/api/people/69/",
      "https://swapi.co/api/people/74/",
      "https://swapi.co/api/people/81/",
      "https://swapi.co/api/people/84/",
      "https://swapi.co/api/people/85/",
      "https://swapi.co/api/people/86/",
      "https://swapi.co/api/people/35/"
    ],
    "films": [
      "https://swapi.co/api/films/2/",
      "https://swapi.co/api/films/7/",
      "https://swapi.co/api/films/5/",
      "https://swapi.co/api/films/4/",
      "https://swapi.co/api/films/6/",
      "https://swapi.co/api/films/3/",
      "https://swapi.co/api/films/1/"
    ],
    "created": "2014-12-10T13:52:11.567000Z",
    "edited": "2015-04-17T06:59:55.850671Z",
    "url": "https://swapi.co/api/species/1/"
  }
];
