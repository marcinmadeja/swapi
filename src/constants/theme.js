import { createMuiTheme } from 'material-ui/styles';
import blueGrey from 'material-ui/colors/blueGrey';
import yellow from 'material-ui/colors/yellow';

const templateStandard = {
  palette: {
    primary: blueGrey,
    secondary: yellow,
    borderColor: '#e6ecf5',
    background: {
      default: '#EDF2F6',
    },
  },
  borderRadius: '5px',
};

const mainTheme = createMuiTheme(templateStandard);

export const gutter = '15px';
export const containerWidth = '1300px';

export default mainTheme;
