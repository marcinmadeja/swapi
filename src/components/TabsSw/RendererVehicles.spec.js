import React from 'react';
import 'jest-styled-components';
import { MemoryRouter } from 'react-router-dom';
import { swapiVehicles } from 'constants/testConstants';
import RendererVehicles from './RendererVehicles';

const loadedData = swapiVehicles;
const urlList = [swapiVehicles[0].url];

const defaultProps = {
  urlList: [],
  loadedData: [],
};

const setup = (props = {}) => {
  props = { ...defaultProps, ...props };
  const component = <RendererVehicles {...props} />;
  const componentWithRouter = <MemoryRouter><RendererVehicles {...props} /></MemoryRouter>;
  const shallowComponent = shallow(component);

  return {
    component,
    componentWithRouter,
    shallowComponent,
  };
};


describe('RendererVehicles', () => {
  it('should render null when empty urlList or empty loadedData', () => {
    const { shallowComponent: emptyUrlList } = setup({ urlList: [], loadedData: [1] });
    const { shallowComponent: emptyLoadedData } = setup({ urlList: [1], loadedData: [] });

    expect(emptyUrlList.html()).toEqual(null);
    expect(emptyLoadedData.html()).toEqual(null);
  });

  it('should match snapshot', () => {
    const { componentWithRouter } = setup({ urlList, loadedData });
    const tree = renderer
      .create(componentWithRouter)
      .toJSON();

    expect(tree).toMatchSnapshot(tree);
  });
});
