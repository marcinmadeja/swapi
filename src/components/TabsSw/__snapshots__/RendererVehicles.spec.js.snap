// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
.c0 {
  margin-bottom: 25px;
  font-size: 18px;
  line-height: 1.5;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  -webkit-transition: 0.33s all;
  transition: 0.33s all;
}

.c0:hover {
  color: #607d8b;
}

.c0 strong {
  font-size: 16px;
}

<a
  className="c0"
  href="/vehicles/snowspeeder/14"
  onClick={[Function]}
>
  Snowspeeder
  <br />
  model: 
  <strong>
    t-47 airspeeder
  </strong>
</a>
`;
